### PR TITLE
+per #15955 Add lower bounds to SnapshotSelectionCriteria

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

**Comment by patriknw** [comment]
Refs #15955

File: akka-persistence/src/main/scala/akka/persistence/SnapshotProtocol.scala [hunk 1]
```diff
@@ -123,6 +135,13 @@ object SnapshotSelectionCriteria {
   /**
    * Java API.
    */
+  def create(maxSequenceNr: Long, maxTimestamp: Long,
+             minSequenceNr: Long, minTimestamp: Long) =
+    SnapshotSelectionCriteria(maxSequenceNr, maxTimestamp, minSequenceNr, minTimestamp)
```
Comment (ktoso) [inline akka-persistence/src/main/scala/akka/persistence/SnapshotProtocol.scala]: Additional Java APIs?

`def create(maxSequenceNr: Long, maxTimestamp: Long) =`?

Comment (patriknw) [inline akka-persistence/src/main/scala/akka/persistence/SnapshotProtocol.scala]: there was already a `def create(maxSequenceNr: Long, maxTimestamp: Long)`
we can't provide all permutations (default values)

Comment (ktoso) [inline akka-persistence/src/main/scala/akka/persistence/SnapshotProtocol.scala]: yeah because all are Long... okey, all good then

**Comment by ktoso** [comment]
LGTM

**Pull request merged**